js/model3D/shared/Canvas3D.svelte

```diff
@@ -52,176 +42,91 @@
 
 	/* BabylonJS engine and scene management */
 	let canvas: HTMLCanvasElement;
-	let scene: BABYLON.Scene;
-	let engine: BABYLON.Engine;
-	let point_cloud_system: BABYLON.PointsCloudSystem | null = null;
+	let viewer: BABYLON_VIEWER.Viewer;
+	let viewerDetails: Readonly<BABYLON_VIEWER.ViewerDetails>;
 	let mounted = false;
 
 	onMount(() => {
-		// Initialize BabylonJS engine and scene
-		engine = new BABYLON.Engine(canvas, true);
-		scene = new BABYLON.Scene(engine);
-
-		scene.createDefaultCameraOrLight();
-		scene.useRightHandedSystem = true;
-		scene.clearColor = scene.clearColor = new BABYLON.Color4(...clear_color);
-
-		engine.runRenderLoop(() => {
-			scene.render();
+		// Initialize BabylonJS viewer
+		BABYLON_VIEWER.createViewerForCanvas(canvas, {
+			clearColor: clear_color,
+			useRightHandedSystem: true,
+			cameraAutoOrbit: { enabled: false },
+			onInitialized: (details: Readonly<BABYLON_VIEWER.ViewerDetails>) => {
+				viewerDetails = details;
+			}
+		}).then((promiseViewer: BABYLON_VIEWER.Viewer) => {
+			viewer = promiseViewer;
+			mounted = true;
 		});
 
-		function onWindowResize(): void {
-			engine.resize();
-		}
-		window.addEventListener("resize", onWindowResize);
-
-		mounted = true;
-
 		return () => {
-			scene.dispose();
-			engine.dispose();
-			window.removeEventListener("resize", onWindowResize);
+			viewer.dispose();
 		};
 	});
 
 	$: mounted && load_model(resolved_url);
 
-	function load_model(url: string | undefined): void {
-		if (scene) {
-			// Dispose of the previous model before loading a new one
-			scene.meshes.forEach((mesh) => {
-				mesh.dispose();
-			});
-
-			if (point_cloud_system) {
-				point_cloud_system.dispose();
-				point_cloud_system = null;
-			}
+	function setRenderingMode(pointsCloud: boolean, wireframe: boolean): void {
```

Comment:
nice! so much cleaner. We can easily add support for the other rendering modes. Could do in another PR

Reply:
Let's split the work accross different PR.